lib/cb/utils/validator.rb

```diff
@@ -23,8 +23,19 @@ def validate(response)
 
       def raise_response_code_errors(response)
         code = response.code rescue nil
-        fail Cb::ServiceUnavailableError if code == 503 || simulate_auth_outage?
-        fail Cb::UnauthorizedError if code == 401
+        fail_with_error_details(response, Cb::ServiceUnavailableError) if code == 503 || simulate_auth_outage?
+        fail_with_error_details(response, Cb::UnauthorizedError) if code == 401
+        fail_with_error_details(response, Cb::DocumentNotFoundError) if code == 404
+        fail_with_error_details(response, Cb::BadRequestError) if code && code >= 400 && code < 500
+        fail_with_error_details(response, Cb::ServerError) if code && code >= 500
```

Comment:
So cool story...sometimes response.code is nil and nil will not respond to > or < but nil is fine being compared to a number